### DISCUSSION TITLE
Lights in office turn off with this plugin (using 2 Hue Motion Sensors)

### DISCUSSION THREAD

**Discussion by gizbug** [Jason-Morcos/homebridge-magic-occupancy#42] [Q&A] [answered]
Have 2 hue motion sensors in my office. Also have 3 Hue bulbs in my overhead light fixture.  When I'm sitting at my desk, I am able to have the lights turn on at night (as I setup in my IOS Home Automation) however after a few minutes, they turn off, and i have to jump all around to get them to turn back on.   Is something not set up correctly?

<img width="785" alt="Screen Shot 2023-05-27 at 7 39 37 PM" src="https://github.com/Jason-Morcos/homebridge-magic-occupancy/assets/70644196/6612c5b6-91b7-4f98-a43b-0b187da802de">

Then, created automations in HomeKit:

Motion Sensor A detects motion > turn on MAGIC1
Motion Sensor B detects motion > turn on MAGIC2
Motion Sensor A does not detect motion > turn off MAGIC1
Motion Sensor B does not detect motion > turn off MAGIC2
Example Occupancy sensor detects occupancy > turn on the light
Example Occupancy sensor does not detect occupancy > turn off the light

**Comment by Jason-Morcos** [comment]
I'm not sure I quite understand - when you say "have them turn on at night" do you mean have them get turned on by the motion-based automations?
If the latter, those settings all look correct, but do look to only keep the lights on for 3 minutes without motion. Is it possible that your motion sensors aren't continuously detecting motion while you're in the office and you're sitting still for three minutes? If so, I might consider upping the stay occupied delay quite a bit.

**Reply by gizbug** [reply]
![IMG_0907](https://github.com/Jason-Morcos/homebridge-magic-occupancy/assets/70644196/08f94220-4ef6-4e75-8075-1a18af78e3a1)
> I'm not sure I quite understand - when you say "have them turn on at night" do you mean have them get turned on by the motion-based automations?
> If the latter, those settings all look correct, but do look to only keep the lights on for 3 minutes without motion. Is it possible that your motion sensors aren't continuously detecting motion while you're in the office and you're sitting still for three minutes? If so, I might consider upping the stay occupied delay quite a bit.

Jason, thanks for the quick reply.
By "have them turn on at night" i mean in my IOS Home automations, for Time of Day, instead of motion sensors turning on Magic1 or Magic2 anytime, I have it set to "at night" as I don't need to office light on during daylight hours.  Sorry did not clarify.  I will up the stay occupied delay to say 10 minutes.  Should I mess with Absolute Max Occupation Time?

**Reply by Jason-Morcos** [reply]
Ahh, if you want the lights to stay on sans motion when turned on by that night time automation, I think you need to add a bit here.
I think you need to set up a "Light Switch" and link it to your actual lights (per the readme) so that the lights stay on (indefinitely) if turned on based on a particular time and/or automatically by time of day

**Reply by gizbug** [reply]
I think I'll just stick with your default setup listed above and just have lights turn on any time of day and skip the advanced "Light Switch" setup.  
Im also unclear as to if I need to define all 6 of your switches on the UI Config page, or just ones I am using?

**Reply by Jason-Morcos** [reply]
> Ok Ill redo it all and setup the "Light Switch"
> 
> Im also unclear as to if I need to define all 6 of your switches on the UI Config page, or just ones I am using?

Shouldn't need to redo it! Just need to add the one switch

You can just fill out the ones you're using

**Reply by gizbug** [reply]
> > Ok Ill redo it all and setup the "Light Switch"
> > Im also unclear as to if I need to define all 6 of your switches on the UI Config page, or just ones I am using?
> 
> Shouldn't need to redo it! Just need to add the one switch
> 
> You can just fill out the ones you're using

Ok. Let me try the following below for 24 hours and see how it works, unless you see something off with this new setup. Thanks again for your time.

{
        "name": "Occupancy",
        "stayOccupiedDelay": 300,
        "maxOccupationTimeout": 86400,
        "occupancyLogging": true,
        "persistBetweenReboots": true,
        "startOnReboot": false,
        "lightSwitchesNames": "Light Switch",
        "statefulSwitchesNames": "Motion Switch",
        "triggerSwitchesNames": "Trigger Switch",
        "statefulStayOnSwitchesNames": "",
        "triggerStayOnSwitchesNames": "",
        "createMasterShutoff": false,
        "accessory": "MagicOccupancy"
    }

HomeKit Automations
When Lights turn on -> Turn on Occupancy Light Switch
When Lights turn off -> Turn off Occupancy Light Switch
When Occupancy Sensor Detects Occupancy -> Turn on Light
When Occupancy Sensor Stops Detecting Occupancy -> Turn off Light
When Motion Sensor 1 Detects Motion (between sunset to sunrise) -> Turn on Occupancy Motion Switch
When Motion Sensor 1 Stops Detecting Motion (between sunset to sunrise) -> Turn off Occupancy Motion Switch
When Motion Sensor 2 Detects Motion (between sunset to sunrise) -> Turn on Occupancy Motion Switch
When Motion Sensor 2 Stops Detecting Motion (between sunset to sunrise) -> Turn off Occupancy Motion Switch


The Motion sensors are my two hue motion sensors.

**Reply by Jason-Morcos** [reply, marked as answer]
That looks good - perhaps I would consider using two separate motion switches for each of the motion sensors however (so that there't not a path for the switch to be turned off when one of the motion sensors is actually still active)
